src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5075,36 +5075,23 @@ public override IOperation VisitAnonymousObjectCreation(IAnonymousObjectCreation
             var initializerBuilder = ArrayBuilder<IOperation>.GetInstance(operation.Initializers.Length);
             for (int i = 0; i < operation.Initializers.Length; i++)
             {
-                IOperation initializer = operation.Initializers[i];
+                var simpleAssignment = (ISimpleAssignmentOperation)operation.Initializers[i];
+                var propertyReference = (IPropertyReferenceOperation)simpleAssignment.Target;
                 IPropertySymbol initializedProperty = properties[i];
 
-                // Due to https://github.com/dotnet/roslyn/issues/25266,
-                // the initializer may or may not be a simple assignment to anonymous type property.
-                if (initializer.Kind == OperationKind.SimpleAssignment)
-                {
-                    var simpleAssignment = (ISimpleAssignmentOperation)initializer;
-                    if (simpleAssignment.Target.Kind == OperationKind.PropertyReference)
-                    {
-                        var propertyReference = (IPropertyReferenceOperation)simpleAssignment.Target;
-
-                        // Due to https://github.com/dotnet/roslyn/issues/22736,
-                        // Instance is null for property reference in an anonymous initializer.
-                        if (propertyReference.Instance == null &&
-                            propertyReference.Property == initializedProperty &&
-                            propertyReference.Arguments.IsEmpty)
-                        {
-                            IOperation visitedTarget = new PropertyReferenceExpression(initializedProperty, propertyReference.Instance, ImmutableArray<IArgumentOperation>.Empty,
-                                semanticModel: null, propertyReference.Syntax, propertyReference.Type, propertyReference.ConstantValue, IsImplicit(propertyReference));
-                            IOperation visitedValue = visitAndCaptureInitializer(initializedProperty, simpleAssignment.Value);
-                            var visitedAssignment = new SimpleAssignmentExpression(visitedTarget, isRef: simpleAssignment.IsRef, visitedValue,
-                                semanticModel: null, simpleAssignment.Syntax, simpleAssignment.Type, simpleAssignment.ConstantValue, IsImplicit(simpleAssignment));
-                            initializerBuilder.Add(visitedAssignment);
-                            continue;
-                        }
-                    }
-                }
+                Debug.Assert(propertyReference != null);
+                Debug.Equals(propertyReference.Property, initializedProperty);
+                Debug.Assert(propertyReference.Arguments.IsEmpty);
+                Debug.Assert(propertyReference.Instance != null);
+                Debug.Assert(propertyReference.Instance.Kind == OperationKind.InstanceReference);
```

Comment:
Is it not possible to have nested anonymous type initializers?

Reply:
For nested initializers, the implicit instance is still the innermost one.

---
In reply to: [196887763](https://github.com/dotnet/roslyn/pull/28005#discussion_r196887763) [](ancestors = 196887763)